singer/messages.py

```diff
@@ -276,3 +278,26 @@ def write_version(stream_name, version):
     write_version(stream, version)
     """
     write_message(ActivateVersionMessage(stream_name, version))
+
+
+def read_tap(name, config, **kwargs):
```

Comment:
I notice that this only works for taps that only take a config, and doesn't support the rest of the standard parameters to taps. Could this be extended to include `--state` and `--catalog`?

`--discover` is a bit different, but should be captured in this style of interface to maximize usefulness. Adding a comment to the whole review on the topic.